File: server_gui/src/com/ibm/ima/resources/data/RunMode.java
/*
 * Copyright (c) 2013-2021 Contributors to the Eclipse Foundation
 * 
 * See the NOTICE file(s) distributed with this work for additional
 * information regarding copyright ownership.
 * 
 * This program and the accompanying materials are made available under the
 * terms of the Eclipse Public License 2.0 which is available at
 * http://www.eclipse.org/legal/epl-2.0
 * 
 * SPDX-License-Identifier: EPL-2.0
 */

package com.ibm.ima.resources.data;

import java.util.Arrays;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * REST representation of run modes. Used to set or get change
 * the run mode of the server.
 *
 */
public class RunMode extends  AbstractIsmConfigObject {

    public static final String COPYRIGHT = "\n\nCopyright (c) 2014-2021 Contributors to the Eclipse Foundation\n" +
        "See the NOTICE file(s) distributed with this work for additional\n" +
        "information regarding copyright ownership.\n\n" +
        "This program and the accompanying materials are made available under the\n" +
        "terms of the Eclipse Public License 2.0 which is available at\n" +
        "http://www.eclipse.org/legal/epl-2.0\n\n" +
        "SPDX-License-Identifier: EPL-2.0\n\n";

	
    @SuppressWarnings("unused")
    private final static String CLAS = RunMode.class.getCanonicalName();

    /**
     * Valid modes available
     *
     */
    public enum ValidModes {
    	
    	PRODUCTION("0"),
    	MAINTENANCE("1"),
    	CLEAN_STORE("2");
    	
		private String modeNumber;


		ValidModes(String modeNumber) {
			this.modeNumber = modeNumber;
		}


		public String getModeNumber() {
			return this.modeNumber;
		}
		
    }

    @JsonProperty("Mode")
    private String Mode;
   

    /**
     * Get the runmode value that will be passed to the
     * server.
     * 
     * @return the run mode
     */
    @JsonProperty("Mode")
    public String getMode() {
    	ValidModes modeValue = ValidModes.valueOf(Mode);
        return modeValue.getModeNumber();
    }
    

    /**
     * Get the string representation of the runmode.
     * 
     * @return A string value that represents the runmode
     */
    @JsonIgnore
    public String getModeString() {
    	return Mode;
    }
    
    /**
     * Set the runmode value
     * 
     * @param Mode the run mode
     */
    @JsonProperty("Mode")
    public void setMode(String Mode) {
        this.Mode = Mode;
    }
    
    
    @Override
	public String toString() {
		return "RunMode [Mode=" + Mode + "]";
	}

    @Override
    @JsonIgnore
    public ValidationResult validate() {
    	
    	// must be valid mode and not empty
        ValidationResult vr = new ValidationResult();
        if (Mode == null || trimUtil(Mode).length() == 0) {
            vr.setResult(VALIDATION_RESULT.VALUE_EMPTY);
            return vr;
        }
        try {
            ValidModes.valueOf(Mode);
        } catch (Exception e) {
            vr.setResult(VALIDATION_RESULT.INVALID_ENUM);
            vr.setParams(new Object[] {Mode, Arrays.toString(ValidModes.values())});
        }

        return vr;
    }

}

File: client_python/scripts/webSocketOriginals/runmode.py
#!/usr/bin/python
# Copyright (c) 2013-2021 Contributors to the Eclipse Foundation
# 
# See the NOTICE file(s) distributed with this work for additional
# information regarding copyright ownership.
# 
# This program and the accompanying materials are made available under the
# terms of the Eclipse Public License 2.0 which is available at
# http://www.eclipse.org/legal/epl-2.0
# 
# SPDX-License-Identifier: EPL-2.0
#

# MessageSight command line interface to query/set RUNMODE

# Import the Message Sight Python Library
import os
path = os.path.dirname(os.path.realpath(__file__))

execfile(path + '/messageSightPythonLib.py')

#------------------------------------------------------------------------------
#
#------------------------------------------------------------------------------
def parseModeArgs( obj_args ):
#    print( 'Entering: '+ func_name + __name__ + ' parseModeArgs ' +'  on args: '+ str(obj_args) )
    trace( '...Entering:  parseModeArgs  on args: '+ str(obj_args) )
#    Really do not expect multiple arguments, but will pass on what is received if it is a known mode.
    imamodes = [ 'production=0', 'maintenance=1', 'clean_store=2' ]
    json_str = ""
    for item in obj_args:
        if '=' in item :
            name, value = item.split( '=', 1 )
            json_str = json_str +', "'+ name +'":"'+ value +'"'
            
            print( 'Name: '+ name +' , Value: '+ value )
        else :
            modecode = [s for s in imamodes if item in s]
            if len(modecode) > 0 :
                trace( 'MODE: '+ str(modecode) )
                name, value = modecode[0].split('=')
                json_str = json_str +',"Mode":"'+ value +'"'
            else:
                print( '\nWARNING!  Unrecognized option: "'+ item +'", it will be passed as is.' )
                json_str = json_str +',"Mode":"'+ item +'"'
                
        
    trace( 'JSON_STR: '+ json_str )
    return ( json_str )    
    
#------------------------------------------------------------------------------
# Main module to process arg and send POST
#------------------------------------------------------------------------------
def main(argv):
    adminServer = ''
    uriPath = '/ima/admin/config'

    # Parse arguments
    parser = argparse.ArgumentParser(description='Query or Set MessageSight RUNMODE')
    parser.add_argument('-s', metavar='adminServerIP:port', required=True, type=str, help='adminServerIP:port, ex: 198.23.126.136:9089')
    parser.add_argument('mode', metavar='Mode', type=str, nargs=argparse.REMAINDER, help='Configuration object type', choices=[ 'production', 'maintenance', 'clean_store'] )

    args = parser.parse_args()

    adminServer = args.s
    trace( 'LEN(args )='+ str(len(sys.argv)) )
    objectType = ""
    if len( sys.argv ) >= 4 :
        objectType = parseModeArgs( args.mode )

    # Set headers
    headers = {"Content-type": "application/x-www-form-urlencoded", "Accept": "text/plain"}

    # Set payload
    # CMD: { "Action":"setmode","User":"root"  [,"Mode":"[0|1|2]" ]}  [Production|Maintenance|Clean_store]
    payload_str = '{"Locale":"en_US","User":"admin","Action":"setmode"'+ str(objectType) + '}'
    trace( 'CMD: '+ payload_str )
    payload = json.loads(payload_str)

    # send POST request
    url = 'http://' + adminServer+uriPath

    try:
        response = requests.post(url=url,data=json.dumps(payload), headers=headers)
    except:
        print('No response from server : "'+ adminServer +'".  CHECK the IP and PORT.' )
        sys.exit(1)

    # Process the response and print return information
    rc = parseResponse( 'Runmode', adminServer, objectType, response )

    return( int(rc) )

    
# Check and invoke main
if __name__ == "__main__":
   main(sys.argv[1:])

